portal-api/src/v1/controllers/deal-status/user-can-submit-deal.js

```diff
@@ -1,5 +1,13 @@
+const { CHECKER, MAKER } = require('../../roles/roles');
+
 const userCanSubmitDeal = (deal, user) => {
-  const isMakerCheckerUser = (user.roles.includes('maker') && user.roles.includes('checker'));
+  const userRoles = user?.roles;
+
+  if (!userRoles) {
+    return false;
+  }
+
+  const isMakerCheckerUser = (userRoles.includes(MAKER) && userRoles.includes(CHECKER));
```

Comment:
Function can be refactored, since this `role` will be depreciated so happy to leave as it is.